File: backend/src/main/java/com/carsharing/backend/model/Ride.java
package com.carsharing.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "rides")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Ride {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String startLocation;

    @Column(nullable = false)
    private String endLocation;

    @Column(nullable = false)
    private LocalDateTime departureTime;

    @Column(nullable = false)
    private Integer totalSeats;

    @Column(nullable = false)
    private Integer availableSeats;

    @Column(nullable = false)
    private BigDecimal pricePerSeat;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "driver_id", nullable = false)
    private User driver;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private RideStatus status = RideStatus.ACTIVE;

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createdAt;

    private String carModel;
    private String carColor;
    private String description;

    public enum RideStatus {
        ACTIVE,      // available ride
        FULL,        // all seats booked
        COMPLETED,   // completed ride
        CANCELLED    // cancelled ride
    }
}

File: frontend/lib/models/ride.dart
class Ride {
  final int id;
  final String startLocation;
  final String endLocation;
  final DateTime departureTime;
  final double pricePerSeat;
  final int availableSeats;
  final int driverId;
  final String driverName;

  Ride({
    required this.id,
    required this.startLocation,
    required this.endLocation,
    required this.departureTime,
    required this.pricePerSeat,
    required this.availableSeats,
    required this.driverId,
    required this.driverName,
  });

  factory Ride.fromJson(Map<String, dynamic> json) {
    return Ride(
      id: json['id'],
      startLocation: json['startLocation'],
      endLocation: json['endLocation'],
      departureTime: DateTime.parse(json['departureTime']),
      pricePerSeat: (json['pricePerSeat'] as num).toDouble(),
      availableSeats: json['availableSeats'],
      driverId: json['driverId'],
      driverName: json['driverName'],
    );
  }
}